logic/sudoku.ts

```diff
@@ -0,0 +1,146 @@
+// Cells to fill in the board
+export const emptyCellCount = 30;
+
+// Function to check if a value can be placed in a given cell
+const isValid = (
+  board: number[][],
+  row: number,
+  col: number,
+  num: number,
+): boolean => {
+  // Check if the number is already in the same row or column
+  for (let i = 0; i < 9; i++) {
+    if (board[row]![i] === num || board[i]![col] === num) {
+      return false;
+    }
+  }
+
+  // Check if the number is already in the 3x3 subgrid
+  const startRow = Math.floor(row / 3) * 3;
+  const startCol = Math.floor(col / 3) * 3;
+  for (let i = startRow; i < startRow + 3; i++) {
+    for (let j = startCol; j < startCol + 3; j++) {
+      if (board[i]![j] === num) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+};
+
+// Function to solve the Sudoku puzzle using backtracking
+const solveSudoku = (board: number[][]): boolean => {
+  for (let row = 0; row < 9; row++) {
+    for (let col = 0; col < 9; col++) {
+      if (board[row]![col] === 0) {
+        for (let num = 1; num <= 9; num++) {
+          if (isValid(board, row, col, num)) {
+            board[row]![col] = num;
+
+            if (solveSudoku(board)) {
+              return true;
+            }
+
+            board[row]![col] = 0;
+          }
+        }
+
+        return false;
+      }
+    }
+  }
+
+  return true; // All cells filled
+};
+
+export const generateSolvedSudoku = (): number[][] => {
```

Comment:
**Q:** Wouldn't that generate the exact same board each time?

Reply:
Nope, I've checked that few times, it does not generate exactly same board - even if it does generate, most of old people will forget the solution xD